File: 2962.java
class Solution {
    public long countSubarrays(int[] nums, int k) {
        int max = Integer.MIN_VALUE;
        for (int i : nums) {
            if (i > max) {
                max = i;
            }
        }
        long result = 0;
        int start = 0, countMaxInWindow = 0;
        for (int end = 0; end < nums.length; end++) {
            if (nums[end] == max) {
                countMaxInWindow++;
            }
            while (k == countMaxInWindow) {
                if (nums[start] == max) {
                    countMaxInWindow--;
                }
                start++;
            }
            result += start;
        }
        return result;
    }
}

File: Python/2962.py
from typing import List


class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        max_element = max(nums)
        max_count = result = left = 0

        for right in range(len(nums)):
            if nums[right] == max_element:
                max_count += 1

            while max_count == k:
                if nums[left] == max_element:
                    max_count -= 1
                left += 1

            result += left

        return result
